server/src/internalClusterTest/java/org/elasticsearch/search/fieldcaps/FieldCapsHasValueIT.java

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.fieldcaps;
+
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.fieldcaps.FieldCapabilities;
+import org.elasticsearch.action.fieldcaps.FieldCapabilitiesResponse;
+import org.elasticsearch.action.index.IndexRequestBuilder;
+import org.elasticsearch.action.support.ActiveShardCount;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST)
+public class FieldCapsHasValueIT extends ESIntegTestCase {
```

Comment:
The amount of test cases in this class is impressive, that's great. I do wish that we had more unit tests, and less integ tests, especially those that extend from ESIntegTestCase as they startup a cluster with multiple nodes, they can be flaky and difficult to reproduce.

Could we try to be explicit about what specific aspects we are testing instead of testing many different aspects e.g. serialization / distributed aspects / functionality / edge cases through integ tests?

Perhaps we should also extend the existing field_caps yaml tests as well to provide the flag. We have multi_cluster tests around it and it would be good to add the flag there too. I think that's going to reveal that we haven't added the new parameter to the rest spec, hence clients cannot provide it :)